src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -2286,48 +2286,78 @@ private IOperation CreateBoundDiscardPatternOperation(BoundDiscardPattern boundN
 
         internal IPropertySubpatternOperation CreatePropertySubpattern(BoundPropertySubpattern subpattern, ITypeSymbol matchedType)
         {
-            SyntaxNode syntax = subpattern.Syntax;
-            IOperation member = CreatePropertySubpatternMember(subpattern.Member, matchedType) ??
-                                OperationFactory.CreateInvalidOperation(_semanticModel, syntax, ImmutableArray<IOperation>.Empty, isImplicit: true);
-            IPatternOperation pattern = (IPatternOperation)Create(subpattern.Pattern);
-            return new PropertySubpatternOperation(member, pattern, _semanticModel, syntax, isImplicit: false);
-        }
+            // We treat `c is { ... .Prop: <pattern> }` as `c is { ...: { Prop: <pattern> } }`
 
-        internal IOperation? CreatePropertySubpatternMember(BoundPropertySubpatternMember? member, ITypeSymbol matchedType)
-        {
+            SyntaxNode subpatternSyntax = subpattern.Syntax;
+            BoundPropertySubpatternMember? member = subpattern.Member;
+            IPatternOperation pattern = (IPatternOperation)Create(subpattern.Pattern);
             if (member is null)
-                return null;
+            {
+                var reference = OperationFactory.CreateInvalidOperation(_semanticModel, subpatternSyntax, ImmutableArray<IOperation>.Empty, isImplicit: true);
+                return new PropertySubpatternOperation(reference, pattern, _semanticModel, subpatternSyntax, isImplicit: false);
+            }
 
-            switch (member.Symbol)
+            // Create an operation for last property access:
+            // `{ SingleProp: <pattern operation> }`
+            // or
+            // `.LastProp: <pattern operation>` portion (treated as `{ LastProp: <pattern operation> }`)
+            var nameSyntax = member.Syntax;
+            var inputType = member.Receiver?.Type.StrippedType().GetPublicSymbol() ?? matchedType;
```

Comment:
>member.Receiver?.Type.StrippedType().GetPublicSymbol() ?? matchedType [](http://example.com/codeflow?start=28&length=69)

It looks like this value is/was shared between the two `createPropertySubpattern()` calls. (Sorry for missing that with my earlier comment.) Perhaps add a local function to calculate the value.